java/flight/flight-core/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -353,6 +361,23 @@ private InputStream asInputStream(BufferAllocator allocator) {
         // the reference count
         b.getReferenceManager().retain();
       }
+
+      // add compression info, if any
+      if (bodyCompression != null) {
+        ArrowBuf compBuf = allocator.buffer(ArrowBodyCompression.BODY_COMPRESSION_LENGTH);
+        compBuf.setByte(0, bodyCompression.getCodec());
```

Comment:
it looks like this is manually serializing the table, shouldn't this be delegated to flatbuffers?

Reply:
In the reivsed PR, we no longer manually serilize/deserialize the table. Thanks.